engine/schema/src/main/java/com/cloud/host/HostVO.java

```diff
@@ -714,9 +715,10 @@ public void setResourceState(ResourceState state) {
 
     @Override
     public boolean isInMaintenanceStates() {
-        return (getResourceState() == ResourceState.Maintenance || getResourceState() == ResourceState.ErrorInMaintenance || getResourceState() == ResourceState.PrepareForMaintenance);
+        ResourceState state = getResourceState();
```

Comment:
Same as above - simplify using object in array, `Arrays.asList().contains()` or equivalent

Reply:
done